wasm-rpc-stubgen/src/model/component.rs

```diff
@@ -0,0 +1,202 @@
+                                    Ok(properties) => Some(properties),
+                                    Err(err) => {
+                                        add_component_error(format!(
+                                            "Failed to get worker RPC trait properties: {}",
+                                            err
+                                        ));
+                                        has_errors = true;
+                                        None
+                                    }
+                                }
+                            }
+                            other => {
+                                eprintln!("Skipping unknown trait: {}", other);
+                                None
+                            }
+                        };
+                        if let Some(properties) = properties {
+                            worker_rpc_dependencies.push(properties.component_name);
+                        }
+                    }
+
+                    (!has_errors).then(|| worker_rpc_dependencies)
+                };
+
+                if let (Some(properties), Some(worker_rpc_dependencies)) =
+                    (properties, worker_rpc_dependencies)
+                {
+                    components.push(Component {
+                        name: component.name,
+                        source: application.source.clone(),
+                        wit: properties.wit.into(),
+                        wasm: properties.wasm.into(),
+                        composed_wasm: properties.composed_wasm.into(),
+                        worker_rpc_dependencies,
+                    })
+                }
+            }
+        }
+
+        if errors.is_empty() {
+            Ok(components)
+        } else {
+            Err(errors)
+        }
+    }
+
+    pub fn from_oam_applications(
+        applications: Vec<oam::ApplicationWithSource>,
+    ) -> Result<Vec<Component>> {
+        let mut all_components = Vec::<Component>::new();
+        let mut errors = Vec::<String>::new();
+
+        for app in applications {
+            match Component::from_oam_application(app) {
+                Ok(app_components) => all_components.extend(app_components),
+                Err(errs) => errors.extend(errs),
+            }
+        }
+
+        if errors.is_empty() {
+            Ok(all_components)
+        } else {
+            Err(errors)
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::model::oam::Application;
+    use assert2::assert;
+
+    #[test]
+    fn deserialize_example_application() {
+        let application: Application = serde_yaml::from_str(
+            r#"
+apiVersion: core.oam.dev/v1beta1
+metadata:
+  name: "App name"
+kind: Application
+spec:
+  components:
+    - name: component-one
+      type: component-durable
+      properties:
+        wit: wit
+        wasm: out/component.wasm
+        composedWasm: out/component-composed.wasm
```

Comment:
I think we should stay away from 'composed' terminology as much as possible. A user may think in terms of 'output' or 'target', but probably not 'composed'.

```suggestion
        output-wasm: out/component.wasm
```

Reply:
renamed, but with using camelCase